src/Infrastructure/Doctrine/ORM/ChannelPriceLogHistoryRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Infrastructure\Doctrine\ORM;
+
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+
+final class ChannelPriceLogHistoryRepository extends EntityRepository implements ChannelPriceLogHistoryRepositoryInterface
+{
+    public function createChannelPricingLogEntryListQueryBuilder(string $productVariantId, string $channelPricingId): QueryBuilder
+    {
+        return $this->createQueryBuilder('o')
+            ->innerJoin('o.channelPricing', 'channelPricing')
+            ->innerJoin('channelPricing.productVariant', 'productVariant')
+            ->andWhere('productVariant = :productVariantId')
+            ->andWhere('channelPricing = :channelPricingId')
+            ->orderBy('o.id', 'DESC')
+            ->setParameter('productVariantId', $productVariantId)
+            ->setParameter('channelPricingId', $channelPricingId)
+        ;
```

Comment:
As we have information about id of channel pricing, there is no need for checking also by product variant. IMO this query builder should be created by channel pricing or by product variant and channel.